application/src/main/java/uk/nhs/hee/tis/revalidation/messages/RabbitMessageListener.java

```diff
@@ -107,4 +108,20 @@ public void receiveUpdateMessageFromMasterDoctorView(
         recommendationViewMapper.mapMasterDoctorViewDtoToRecommendationView(
             masterDoctorViewDto));
   }
+
+  /**
+   * disconnect Doctors from a Designated Body (DB) when the GMC give us a DB without them.
+   */
+  @RabbitListener(queues = "${app.rabbit.reval.queue.doctorsfordb.collected.recommendation}")
+  public void disconnectDoctorsFromDbWhenTheyAreNotInGmc(
+      final DoctorsForDbCollectedEvent doctorsForDbCollectedEvent) {
+    try {
+      log.debug("DoctorsForDbCollectedEvent message received from rabbit: {}",
+          doctorsForDbCollectedEvent);
+      doctorsForDBService.disconnectDoctorsFromDb(doctorsForDbCollectedEvent);
+    } catch (Exception exception) {
+      log.warn("Rejecting message for failed doctor update", exception);
+      throw new AmqpRejectAndDontRequeueException(exception);
```

Comment:
I know this is how we've handled failed messages in other listeners (particularly big sync jobs) but not sure if we want to handle it differently in this case, @JosephKelly ?

Reply:
Yeah.  I had a bit of a concern about errors being thrown because of deadlocks / concurrent update exceptions. In those cases, retrying seems like the right thing to do and has a reasonable chance of succeeding.

Reply:
I couldn't handle this comment as I did not see any similar stuff in Recommendation service.